### PR TITLE
Add SmallRye Config 1.8 Blog Post.

### PR DESCRIPTION
**Pull request by radcortez**

### CODE REVIEW

File: _posts/2020-05-22-smallrye-config-1-8.adoc [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+----
+public interface ConfigSourceInterceptor extends Serializable {
+    ConfigValue getValue(ConfigSourceInterceptorContext context, String name);
+}
+----
+
+The `ConfigValue` objects hold information about the key name, value, config source origin, and ordinal.
+
+Additionally, the `ConfigSourceInterceptor` may also intercept resolution of configuration names of configuration
+`ConfigValue` values with the methods `Iterator<String> iterateNames(ConfigSourceInterceptorContext context)` and
```
Comment (Ladicek) [inline _posts/2020-05-22-smallrye-config-1-8.adoc:39]: "resolution of configuration names of configuration `ConfigValue` values" is hard to follow. Would "resolution of configuration names" be enough?

Actually looking at what follows, it seems it's mainly a typo -- "of" instead of "or". I'd still get rid of the "`ConfigValue`" word -- just "resolution of configuration names or configuration values".

File: _posts/2020-05-22-smallrye-config-1-8.adoc [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+== Built-In Interceptors
+
+SmallRye Config provides the following built-in Interceptors to expand its feature set:
+
+* RelocateConfigSourceInterceptor
+* ProfileConfigSourceInterceptor
+* ExpressionConfigSourceInterceptor
+* FallbackConfigSourceInterceptor
+* LoggingConfigSourceInterceptor
+* SecretKeyConfigSourceInterceptor
```
Comment (Ladicek) [inline _posts/2020-05-22-smallrye-config-1-8.adoc:79]: Use backticks on each line `to show the names like this`?

File: _posts/2020-05-22-smallrye-config-1-8.adoc (hunk 1 above)
Comment (radcortez) [reply _posts/2020-05-22-smallrye-config-1-8.adoc]: Yeah it was a typo :)

**Comment by radcortez** [comment]
@Ladicek Thank you for the review. Let me know if this is ok now :)

**Comment by radcortez** [comment]
Ok, let me remove these.

File: _posts/2020-05-22-smallrye-config-1-8.adoc [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+:page-title: SmallRye Config 1.8 Released
+:page-synopsis: SmallRye Config 1.8 Released
+:page-tags: [microprofile,smallrye,config]
+:page-date: 2020-05-22 13:00:00.000 +0100
+:page-author: radcortez
+
+SmallRye Config version 1.8 was released a few days ago.
+
+The latest version of SmallRye Config introduced a few experimental changes to enhance the API and to make SmallRye
+Config more extensible. One of the experimental changes is the new Interceptor model, to intercept the lookup of a
```
Comment (Ladicek) [inline _posts/2020-05-22-smallrye-config-1-8.adoc:12]: Capital "Interceptor" intentional?

File: _posts/2020-05-22-smallrye-config-1-8.adoc [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        return configValue;
+    }
+}
+----
+
+SmallRye Config does ship with its own logging interceptor, so you don't need to implement one.
+
+== Built-In Interceptors
+
+SmallRye Config provides the following built-in Interceptors to expand its feature set:
```
Comment (Ladicek) [inline _posts/2020-05-22-smallrye-config-1-8.adoc:72]: Capital "Interceptors" intentional?

File: _posts/2020-05-22-smallrye-config-1-8.adoc [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+* `ExpressionConfigSourceInterceptor`
+* `FallbackConfigSourceInterceptor`
+* `LoggingConfigSourceInterceptor`
+* `SecretKeyConfigSourceInterceptor`
+
+Not every interceptor is registered by default. Only the `ProfileConfigSourceInterceptor`,
+`ExpressionConfigSourceInterceptor` and `SecretKeyConfigSourceInterceptor` are added on new instances of `Config`.
+Other interceptors require manual registration via the `ServiceLoader` mechanism.
+
+To learn more about all the build-in interceptors, please refer to the
```
Comment (Ladicek) [inline _posts/2020-05-22-smallrye-config-1-8.adoc:85]: "build-in" -> "built-in"?

File: _posts/2020-05-22-smallrye-config-1-8.adoc [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+https://smallrye.io/docs/smallrye-config/interceptors/interceptors.html#_built_in_interceptors[documentation page].
+
+=== Expression Expansion
+
+The `ExpressionConfigSourceInterceptor` provides expression expansion on configuration values. An expression string is
+a mix of plain strings and expression segments, which are wrapped by the sequence `${ ... }`.
```
Comment (Ladicek) [inline _posts/2020-05-22-smallrye-config-1-8.adoc:91]: "wrapped by the sequence" -> perhaps just "wrapped into"?

File: _posts/2020-05-22-smallrye-config-1-8.adoc [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+For instance, the following configuration properties file:
```
Comment (Ladicek) [inline _posts/2020-05-22-smallrye-config-1-8.adoc:93]: Incomplete sentence. Perhaps "For example, consider the following ..."?

File: _posts/2020-05-22-smallrye-config-1-8.adoc [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+name and a dot:
+
+[source,properties]
+----
+my.prop=1234
+%dev.my.prop=5678
+----
+
+Lookup is always performed with the `my.prop` property name. To use the profile `dev`, the configuration
+`smallrye.config.profile=dev` has to be set into any valid `ConfigSource`.
```
Comment (Ladicek) [inline _posts/2020-05-22-smallrye-config-1-8.adoc:124]: "has to be set into any valid `ConfigSource`" looks weird, perhaps just "has to be set in any valid `ConfigSource`"?

File: _posts/2020-05-22-smallrye-config-1-8.adoc [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+When looking up the property `my.prop` with the `dev` profile active the value is `5678`.
```
Comment (Ladicek) [inline _posts/2020-05-22-smallrye-config-1-8.adoc:126]: Comma between "... profile active, the value is ..."?

File: _posts/2020-05-22-smallrye-config-1-8.adoc [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+
+Only one profile can be active at any given time.
+
+== API Enhancements
+
+Aside from interceptors, a few additional API Enhancements shipped with the latest version of SmallRye Config:
```
Comment (Ladicek) [inline _posts/2020-05-22-smallrye-config-1-8.adoc:132]: Capital "Enhancements" intentional?

File: _posts/2020-05-22-smallrye-config-1-8.adoc [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+configuration.
+* `ConfigValueConfigSource` - Extends the original `ConfigSource` to expose methods that return a `ConfigValue`.
+* `SmallRyeConfigBuilder#withSecretKeys` - Hide configuration properties that contain passwords or other kinds of
+secrets to prevent accidental exposure of such values.
+
+== Summary
+
+All the new features detailed here are experimental. The team is happy with them, and they had careful consideration
+when designed. We cannot guarantee that they won't suffer any changes in the next few releases, especially considering
+that the SmallRye team is also pushing to have these added to the MicroProfile API specification, which may require
```
Comment (Ladicek) [inline _posts/2020-05-22-smallrye-config-1-8.adoc:144]: "MicroProfile API specification" -> "MicroProfile Config specification"?

**Comment by radcortez** [comment]
Thanks! Fixed :)
